swagger_server/itm/ta1/adept_ta1_controller.py

```diff
@@ -20,22 +22,42 @@ class AdeptTa1Controller(ITMTa1Controller):
 
     cfg = ITMTa1Controller.config[ITMTa1Controller.config_group]
 
-    ADEPT_EVAL_FILENAMES = generate_list(cfg['ADEPT_EVAL_FILENAMES'])
-    ADEPT_TRAIN_FILENAMES = generate_list(cfg['ADEPT_TRAIN_FILENAMES'])
+    scenario_directory = cfg['SCENARIO_DIRECTORY']
+    try:
+        scenario_files = set(os.listdir(scenario_directory))
+    except OSError:
+        logging.fatal("Invalid filepath. Please check the SCENARIO_DIRECTORY variable in the config.ini file.")
+        scenario_files = set()
+
+    scenario_ids = load_scenario_ids(scenario_directory, scenario_files)
+    alignment_ids = set() if testingMode else load_alignment_ids(ITMTa1Controller.get_contact_info('adept'), '/api/v1/alignment_target_ids')
+
+    ADEPT_EVAL_FILENAMES = sorted(resolve_tokens(cfg['ADEPT_EVAL_FILENAMES'], scenario_files))
+    ADEPT_TRAIN_FILENAMES = sorted(resolve_tokens(cfg['ADEPT_TRAIN_FILENAMES'], scenario_files))
 
     for key, value in cfg.items():
         if scenarioMatch := scenarioRegex.match(key):
             mode = scenarioMatch.group(1).lower()
             group = scenarioMatch.group('group').lower()
             correct_dict = evaluationScenarios if mode == 'eval' else trainingScenarios
-            correct_dict[group] = set(generate_list(value))
+            # Do not sort sets into ordered lists for scenario IDs.
+            # These sets are used in the get_target_ids() method,
+            # and maintaining them as sets allows for O(1) lookup time.
```

Comment:
Remove the statement about O(1) here since performance should not an issue for the number of scenarios we'd be even considering in our wildest dreams.